### PR TITLE
docs(Event Framework): updating and extending doc

### LINKED ISSUE #1970: Update Documentation on event filtering and add missing javadoc to Event Classes
# Feature Request

The documentation about how to filter Events needs to be adapted. Also the events type Assets and TransferProcess needs to get javadoc.

## Which Areas Would Be Affected?
Event Framework

## Why Is the Feature Desired?
Documentation needs always to be up to date.

## Solution Proposal
Adding and adapt documentation

## Type of Issue
Improvement

## Checklist

- [x] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

### PR DESCRIPTION
**Pull request by janpmeyer**
## What this PR changes/adds

Adding missing javadoc and updating documentation about how to filter Events.

## Why it does that

Improved Documentation is always needed

## Linked Issue(s)

Closes #1970

## Checklist

- [ ] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [ ] documented public classes/methods?
- [x] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Comment by jimmarino** [comment]
Before we go down the path of building more on the payload/even distinction, can we have a discussion about whether we want the latter at all?

File: docs/developer/events.md [hunk 1]
```diff
@@ -41,12 +41,17 @@ public class ExampleEventSubscriber implements EventSubscriber {
...
+in addition to filtering on a specific event, such as TransferProcessCompleted, it is possible to react to a group of events 
+that generally have to do with Assets, ContractDefinition, ContractNegotiation, PolicyDefinition and TransferProcess.
+
 ```java
 public class ExampleEventSubscriber implements EventSubscriber {
     
     public void on(Event event) {
-        if (event instanceof AssertCreated) {
+        if (event.getPayload() instanceof TransferProcessEventPayload) {
             // react only to AssertCreated events
```
Comment (ndr-brt) [inline docs/developer/events.md:55]: the comment should be updated as well

**Review by ndr-brt — CHANGES_REQUESTED** [review]

**Comment by janpmeyer** [comment]
@jimmarino 

Sure. I just made this PR just to correct/update the existing documentation. I'm not working on anything new.

File: docs/developer/events.md (hunk 1 above)
Comment (janpmeyer) [reply docs/developer/events.md]: Updated it. Sorry, overread this comment.

File: docs/developer/events.md [hunk 2]
```diff
@@ -41,13 +41,18 @@ public class ExampleEventSubscriber implements EventSubscriber {
...
+in addition to filtering on a specific event, such as TransferProcessCompleted, it is possible to react to a group of events 
+that generally have to do with Assets, ContractDefinition, ContractNegotiation, PolicyDefinition and TransferProcess.
+
 ```java
 public class ExampleEventSubscriber implements EventSubscriber {
     
     public void on(Event event) {
-        if (event instanceof AssertCreated) {
-            // react only to AssertCreated events
+        if (event.getPayload() instanceof TransferProcessEventPayload) {
```
Comment (ndr-brt) [inline docs/developer/events.md:54]: This actually can be done in both ways, filtering by event class type or by payload as well, I would keep both the examples

Comment (janpmeyer) [reply docs/developer/events.md:59]: Added it back. Hope its fine.

**Review by ndr-brt — APPROVED** [review]
I'd say to merge this as it is well describing the current situation. An eventual change of the hierarchy would result in a change of the documentation as well. wdyt @jimmarino ?

**Pull request merged**